docs/tutorials/dataset_creation/custom_space_serialization.py

```diff
@@ -0,0 +1,156 @@
+# fmt: off
+"""
+Serializing a custom space
+=========================================
+"""
+# %%%
+# In this tutorial you'll learn how to serialize a custom Gymnasium observation space and use that
+# to create a Minari dataset with :class:`minari.DataCollectorV0`. We'll use the
+# `MiniGrid Empty <https://minigrid.farama.org/environments/minigrid/EmptyEnv/>`_ environment and
+# show how to serialize its unique observation space.
+#
+# Serializing a custom space can be applied to both observation and action spaces.
+#
+# Let's start by installing the minigrid library:
+#
+# ``pip install minigrid``
+
+# %% [markdown]
+# Then we can import the required modules:
+
+# %%
+import json
+from typing import Dict, Union
+
+import gymnasium as gym
+from minigrid.core.mission import MissionSpace
+
+import minari
+from minari import DataCollectorV0
+from minari.serialization import deserialize_space, serialize_space
+
+
+# %% [markdown]
+# First we'll initialize the MiniGrid Empty environment and take a look at its observation and action space.
+
+# %%
+env = gym.make("MiniGrid-Empty-16x16-v0")
+
+print(f"Observation space: {env.observation_space}")
+print(f"Action space: {env.action_space}")
+
+# %% [markdown]
+# We can see the output from above looks like:
+#
+#   Observation space: Dict('direction': Discrete(4), 'image': Box(0, 255, (7, 7, 3), uint8), 'mission': MissionSpace(<function EmptyEnv._gen_mission at 0x12253a940>, None))
+#
+#   Action space: Discrete(7)
+#
+# If we take a look at Minari's `serialization functions <https://github.com/Farama-Foundation/Minari/blob/main/minari/serialization.py#L13>`_
+# we can see that ``Dict``, ``Discrete``, and ``Box`` are all supported. However ``MissionSpace`` is not
+# supported and if try to serialize it with:
+
+# %%
+serialize_space(env.observation_space['mission'])
```

Comment:
It is better to not create code that throws error, as we aim to add automatic testing for docs (so the file should run). You can catch the error and print it or just use text to explain it.